libtelemetry/src/main/java/com/mapbox/services/android/telemetry/MapboxTelemetry.java

```diff
@@ -277,4 +278,29 @@ public void onServiceDisconnected(ComponentName className) {
       serviceBound = false;
     }
   };
+
+  private boolean checkLocationPermission() {
+    if (PermissionsManager.areLocationPermissionsGranted(context)) {
+      return true;
+    } else {
+      permissionBackoff();
+      return false;
+    }
+  }
+
+  private void permissionBackoff() {
+    Handler handler = new Handler();
```

Comment:
There's no need to create a handler and a counter here because we're already doing it in `PermissionCheckRunnable` (it may be interesting to pass them in through the constructor though).